pkg/core/app/app.go

```diff
@@ -65,11 +66,32 @@ type Options struct {
 	DockerNetwork string
 }
 
+func findComposeFilePathFromCmdArgs(args []string) string {
+	for i := 0; i < len(args); i++ {
+		if args[i] == "-f" && i+1 < len(args) {
+			return args[i+1]
+		}
+	}
+	return ""
+}
+
 func (a *App) Setup(_ context.Context) error {
 
 	if utils.IsDockerKind(a.kind) && isDetachMode(a.logger, a.cmd, a.kind) {
 		return fmt.Errorf("application could not be started in detached mode")
 	}
+	var composeFilePath string
+	var cmdArgs []string
```

Comment:
All logic related to the compose file should be in `SetupCompose` function only.